samples/gen-js/index.js

```diff
@@ -0,0 +1,391 @@
+    return data.join(",");
+  }
+  return data;
+}
+
+module.exports = class SwaggerTest {
+
+  address;
+  timeout;
+
+  constructor(options) {
+    options = options || {};
+
+    if (options.discovery) {
+      this.address = discovery("swagger-test", "default").url();
+    } else if (options.address) {
+      this.address = options.address;
+    } else {
+      throw new Error("Cannot initialize swagger-test without discovery or address");
+    }
+    if (options.timeout) {
+      this.timeout = options.timeout
+    }
+  }
+
+  
+  getBooks = (params, options, cb) => {
+
+    if (!cb && typeof options === "function") {
+      cb = options;
+      options = undefined;
+    }
+
+    if (!options) {
+      options = {};
+    }
+
+    const timeout = options.timeout || this.timeout;
+    const span = options.span;
+
+    const headers = {
+
+    }
+
+    if (span) {
+      opentracing.inject(span, opentracing.FORMAT_TEXT_MAP, headers);
+      span.logEvent("GET /v1/books");
+    }
+
+    const requestOptions = {
+      method: "GET",
+      uri: this.address + "/v1/books",
+      json: true,
+      timeout,
+      headers,
+      body: {
+
+      },
+      qs: {
+
+        "authors": serializeQueryString(params.authors),
+
+        "available": serializeQueryString(params.available),
+
+        "state": serializeQueryString(params.state),
+
+        "published": serializeQueryString(params.published),
+
+        "snake_case": serializeQueryString(params.snakeCase),
+
+        "completed": serializeQueryString(params.completed),
+
+        "maxPages": serializeQueryString(params.maxPages),
+
+        "min_pages": serializeQueryString(params.minPages),
+
+        "pagesToTime": serializeQueryString(params.pagesToTime),
+
+      },
+    };
+    return new Promise((resolve, reject) => {
+      const rejecter = (err) => {
+        reject(err);
+        if (cb) {
+          cb(err);
+        }
+      }
+      const resolver = (data) => {
+        resolve(data);
+        if (cb) {
+          cb(null, data);
+        }
+      }
+
+      request(requestOptions, (err, response, body) => {
+        if (err) {
+          return rejecter(err);
+        }
+        if (response.statusCode >= 400) {
+          return rejecter(new Error(body));
+        }
+        resolver(body);
+      });
+    });
+  }
+
+  
+  getBookByID = (params, options, cb) => {
+
+    if (!cb && typeof options === "function") {
+      cb = options;
+      options = undefined;
+    }
+
+    if (!options) {
+      options = {};
+    }
+
+    const timeout = options.timeout || this.timeout;
+    const span = options.span;
+
+    const headers = {
+
+        "authorization": params.authorization,
+
+    }
+
+    if (span) {
+      opentracing.inject(span, opentracing.FORMAT_TEXT_MAP, headers);
+      span.logEvent("GET /v1/books/{book_id}");
+    }
+
+    const requestOptions = {
+      method: "GET",
+      uri: this.address + "/v1/books/" + params.bookID + "",
```

Comment:
Can't sadly since go doesn't let you escape backticks inside of string literals and I didn't think working around that flaw was worth the benefit.